src/NHibernate/Dialect/Function/CastFunction.cs

```diff
@@ -87,23 +87,27 @@ public SqlString Render(IList args, ISessionFactoryImplementor factory)
 				throw new QueryException(string.Format("invalid Hibernate type for cast(): type {0} not found", typeName));
 			}
 
-			if (CastingIsRequired(sqlType))
-			{
-				return new SqlString("cast(", args[0], " as ", sqlType, ")");
-			}
-			else
-			{
-				return new SqlString("(", args[0], ")");
-			}
+#pragma warning disable 618
+			return CastingIsRequired(sqlType)
+#pragma warning restore 618
+				? Render(args, sqlType, factory)
+				: new SqlString("(", args[0], ")");
```

Comment:
To simplify 6.0 code clean up I think we should try to put full obsolete branch of code to `#pragma` block.
Something like:
```C#
//TODO 6.0: Remove pragma block with it's content
#pragma warning disable 618
if (!CastingIsRequired(sqlType))
    return new SqlString("(", args[0], ")");
#pragma warning restore 618

return Render(args, sqlType, factory);
```

Reply:
Done as suggested.